Release 1.8.0 (#2380)
apollographql/router: CHANGELOG.md

This project adheres to [Semantic Versioning v2.0.0](https://semver.org/spec/v2.0.0.html).

# [1.8.0] - 2023-01-11

## 📃 Configuration

Configuration changes will be [automatically migrated on load](https://www.apollographql.com/docs/router/configuration/overview#upgrading-your-router-configuration). However, you should update your source configuration files as these will become breaking changes in a future major release.

> ### Defer support GA docs and config ([Issue #2368](https://github.com/apollographql/router/issues/2368))

Author: ```suggestion
### Defer support graduates from preview ([Issue #2368](https://github.com/apollographql/router/issues/2368))
```

### Refactor YAML validation error reports ([Issue #2180](https://github.com/apollographql/router/issues/2180))

YAML configuration file validation prints a report of the errors it encountered, but that report was missing some details and occasionally had its diagnostics cursor pointing at the wrong character/line.  It now points at the correct place more reliably.

By [@Geal](https://github.com/geal) in https://github.com/apollographql/router/pull/2347

# [1.7.0] - 2022-12-22

## 🚀 Features

### The Helm chart has graduated from prerelease to general availability ([PR #2380](https://github.com/apollographql/router/pull/2380))

As part of this release, we have promoted the Helm chart from its prerelease "release-candidate" stage to a "stable" version number.  We have chosen to match the version of the Helm chart to the Router version, which is very agreeable with our automated Router releasing pipeline.  This means the first stable version of the Helm chart will be `1.8.0` which will pair with Router 1.8.0 and subsequent versions will be in lock-step.

By [@abernix](https://github.com/abernix) in https://github.com/apollographql/router/pull/2380

> 

Author: Doh.. Resolved in d17fb494.   Good eye.

This project adheres to [Semantic Versioning v2.0.0](https://semver.org/spec/v2.0.0.html).

> # [1.8.0] - 2023-01-11
> 
> # [1.8.0] (unreleased) - 2022-mm-dd
> 

Author: ```suggestion
# [1.8.0] - 2023-01-11
```

### Defer support GA docs and config ([Issue #2368](https://github.com/apollographql/router/issues/2368))

`@defer` has reached GA, however there were still mentions of it being in preview. These have now been removed.

If you had disabled defer support via config then you will need to update your router.yaml

Before:
```yaml
supergraph:
  preview_defer_support: true
```

After:
```yaml
supergraph:
  defer_support: true
```

By [@bryncooke](https://github.com/bryncooke) in https://github.com/apollographql/router/pull/2378

### Remove timeout from otlp exporter ([Issue #2337](https://github.com/apollographql/router/issues/2337))

`batch_processor` configuration contains timeout, so the existing timeout property has been removed from the parent configuration element.

Before:
```yaml
telemetry:
  tracing:
    otlp:
      timeout: 5s
```
After:
```yaml
telemetry:
  tracing:
    otlp:
      batch_processor:
        timeout: 5s
```

By [@bryncooke](https://github.com/bryncooke) in https://github.com/apollographql/router/pull/2338

## 🚀 Features

### Add cache hit/miss metrics ([Issue #1985](https://github.com/apollographql/router/issues/1985))

Add several metrics around the cache.

> Each cache metrics it contains `kind` attribute to know what kind of cache it was (`query planner`, `apq`, `introspection`)

Reviewer: ```suggestion
Each cache metric contains a `kind` attribute to indicate what kind of cache it was (`query planner`, `apq`, `introspection`)
```

> and the `storage` attribute to know where the cache is coming from.

Reviewer: ```suggestion
and a `storage` attribute to indicate where the cache is coming from e.g memory/disk.
```

> `apollo_router_cache_hit_count` to know when it hits the cache.
> 
> `apollo_router_cache_miss_count` to know when it misses the cache.
> 
> `apollo_router_cache_hit_time` to know how much time it takes when it hits the cache.
> 
> `apollo_router_cache_miss_time` to know how much time it takes when it misses the cache.

Reviewer: ```suggestion
The following buckets are exposed:
`apollo_router_cache_hit_count` - cache hits.

`apollo_router_cache_miss_count` - cache misses.

`apollo_router_cache_hit_time` - cache hit duration.

`apollo_router_cache_miss_time` - cache miss duration.
```

apollo_router_cache_miss_time_bucket{kind="query planner",service_name="apollo-router",storage="memory",le="0.4"} 1
apollo_router_cache_miss_time_bucket{kind="query planner",service_name="apollo-router",storage="memory",le="0.5"} 1
apollo_router_cache_miss_time_bucket{kind="query planner",service_name="apollo-router",storage="memory",le="1"} 1
apollo_router_cache_miss_time_bucket{kind="query planner",service_name="apollo-router",storage="memory",le="5"} 1
apollo_router_cache_miss_time_bucket{kind="query planner",service_name="apollo-router",storage="memory",le="10"} 1
apollo_router_cache_miss_time_bucket{kind="query planner",service_name="apollo-router",storage="memory",le="+Inf"} 1
apollo_router_cache_miss_time_sum{kind="query planner",service_name="apollo-router",storage="memory"} 0.000186783
apollo_router_cache_miss_time_count{kind="query planner",service_name="apollo-router",storage="memory"} 1
```

By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2327

### Add support for single instance Redis ([Issue #2300](https://github.com/apollographql/router/issues/2300))

> For `experimental_cache` with redis caching it now works with only a single Redis instance if you provide only one URL.

Reviewer: ```suggestion
Experimental caching via Redis now works with single Redis instances when configured with a single URL.
```

By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2310

### Support TLS connections to single instance Redis ([Issue #2332](https://github.com/apollographql/router/issues/2332))

> TLS support is mandatory for some hosted Redis providers.

Reviewer: ```suggestion
TLS connections are now supported when connecting to single Redis instances. It is useful for connecting to hosted  Redis providers where TLS is mandatory.
```
